h3-widgets/index.ts

```diff
@@ -0,0 +1,213 @@
+  const formula = await ws.getFormula({
+    column: selectedVariable,
+    operation: 'sum',
+    spatialFilter: getSpatialFilterFromViewState(viewState),
+    viewState,
+  })
+  formulaWidget.textContent = Intl.NumberFormat('en-US', {
+    maximumFractionDigits: 0,
+    // notation: 'compact'
+  }).format(formula.value)
+}
+
+const HISTOGRAM_WIDGET_ID = 'urbanity_widget';
+
+async function renderHistogram(ws: WidgetSource) {
+  histogramWidget.parentElement?.querySelector('.loader')?.classList.toggle('hidden', false);
+  histogramWidget.classList.toggle('hidden', true);
+
+  const categories = await ws.getCategories({
+    column: 'urbanity',
+    operation: 'count',
+    filterOwner: HISTOGRAM_WIDGET_ID,
+    spatialFilter: getSpatialFilterFromViewState(viewState),
+    viewState,
+  })
+
+  histogramWidget.parentElement?.querySelector('.loader')?.classList.toggle('hidden', true);
+  histogramWidget.classList.toggle('hidden', false);
+
+  const selectedCategory = filters['urbanity']?.[FilterType.IN]?.values[0];
+  const colors = categories.map((c) => c.name === selectedCategory ? 'rgba(255, 99, 132, 0.2)' : 'rgba(54, 162, 235, 0.2)')
+
+  if (histogramChart) {
+    histogramChart.data.labels = categories.map((c) => c.name);
+    histogramChart.data.datasets[0].data = categories.map((c) => c.value);
+    histogramChart.data.datasets[0].backgroundColor = colors;
+    histogramChart.update();
+  } else {
+    histogramChart = new Chart(histogramWidget, {
+      type: 'bar',
+      data: {
+        labels: categories.map((c) => c.name),
+        datasets: [
+          {
+            label: 'Urbanity category',
+            data: categories.map((c) => c.value),
+            backgroundColor: colors,
+          }
+        ]
+      },
+      options: {
+        onClick: async (ev, elems) => {
+          console.log(elems[0])
+          const index = elems[0]?.index
+          const categoryName = categories[index]?.name
+          if (!categoryName || categoryName === selectedCategory) {
+            removeFilter(filters, { column: 'urbanity' })
+          } else {
+            addFilter(filters, {
+              column: 'urbanity',
+              type: FilterType.IN,
+              values: [categoryName],
+              owner: HISTOGRAM_WIDGET_ID,
+            })
+          }
+          await render()
+        }
+      },
+    })
+  }
+}
+
+const debouncedRenderWidgets = debounce(renderWidgets, 500);
+
+// Main execution
+const map = new maplibregl.Map({
+  container: 'map',
+  style: BASEMAP.DARK_MATTER,
+  interactive: false
+});
+
+const deck = new Deck({
+  canvas: 'deck-canvas',
+  initialViewState: viewState,
+  controller: true
+});
+deck.setProps({
+  onViewStateChange: (props) => {
+    const {longitude, latitude, ...rest} = props.viewState;
+    map.jumpTo({center: [longitude, latitude], ...rest});
+    viewState = props.viewState;
+    debouncedRenderWidgets()
+  }
```

Comment:
Not sure if this is a 'bug' anywhere or just expected, but I notice if we zoom in enough and the spatialFiltersResolution increases to be > 8, we get an error in the console:

> Error: Spatial filters resolution (9) should be less or equal to the spatial index resolution (8)

Maybe we just want to limit the zoom on the map? I think that error is probably expected on the API and API client.

Reply:
This is happening on the quadbin example too with quadbin resolution > 14
I guess this is an actual bug

Reply:
Updated a commit with the latest alpha `0.4.2-alpha.0` that now fixes this

Reply:
Hm - I think I'm still seeing the same error (pasted above) after zooming in enough. I only tested the h3 example and not the quadbin example for this though.

Reply:
Ok I checked again, and got to the max zoom level that the map will allow without this error being shown. Are you sure you are using version `0.4.2-alpha.0` of `@carto/api-client` ?

Reply:
Ah! Yes that might have been it, I think I'd pulled the newest changes but not re-installed dependencies. Not seeing the zoom error anymore. 👍🏻